connector/dialect.tdd

```diff
@@ -380,60 +151,66 @@
       <argument type='datetime' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>(CASE
-	WHEN %1 THEN 1.0
-	WHEN NOT %1 THEN 0.0
-	ELSE NULL END)</formula>
-      <argument type='bool' />
+      <formula>CAST(%1 AS DECIMAL)</formula>
+      <argument type='real' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>CAST(%1 AS DOUBLE PRECISION)</formula>
-      <argument type='real' />
+      <formula>CAST(%1 AS DECIMAL)</formula>
```

Comment:
In a previous comment, I noted that "FLOAT" should probably map to the MongoSQL `DOUBLE` type instead of `DECIMAL`. Regardless, for casting to `DECIMAL` _or_ `DOUBLE`, we actually support the argument being a datetime since we follow mongodb's `$convert` semantics which do support this for [casting to decimal](https://docs.mongodb.com/manual/reference/operator/aggregation/convert/#converting-to-a-decimal) and for [casting to double](https://docs.mongodb.com/manual/reference/operator/aggregation/convert/#converting-to-a-double). So we should add a case for when the argument type is `datetime`.